evalml/data_checks/target_leakage_data_check.py

```diff
@@ -13,23 +13,49 @@
 class TargetLeakageDataCheck(DataCheck):
     """Check if any of the features are highly correlated with the target."""
 
-    def __init__(self, pct_corr_threshold=0.95):
+    def __init__(self, pct_corr_threshold=0.95, pearson_corr=False):
         """Check if any of the features are highly correlated with the target.
 
-        Currently only supports binary and numeric targets and features.
+        If `pearson_corr=False`, supports all target and feature types. Otherwise, only supports binary with numeric and boolean dtypes.
+        Pearson correlation returns a value in [-1, 1], while mutual information returns a value in [0, 1].
 
         Arguments:
             pct_corr_threshold (float): The correlation threshold to be considered leakage. Defaults to 0.95.
+            pearson_corr (bool): Whether or not to use the Pearson Correlation versus mutual information. Defaults to False, which uses mutual info.
 
         """
         if pct_corr_threshold < 0 or pct_corr_threshold > 1:
             raise ValueError("pct_corr_threshold must be a float between 0 and 1, inclusive.")
         self.pct_corr_threshold = pct_corr_threshold
+        self.pearson = pearson_corr
+
+    def _calculate_pearson(self, X, y, messages):
+        highly_corr_cols = []
+        X = X.select_dtypes(include=numeric_and_boolean_dtypes)
+        if len(X.columns) > 0:
+            highly_corr_cols = [label for label, col in X.iteritems() if abs(y.corr(col)) >= self.pct_corr_threshold]
+        return highly_corr_cols
+
+    def _calculate_mutual_information(self, X, y):
+        highly_corr_cols = []
+        target = 'target'
+        while target in X.columns:
+            target += '0'
+        combined = X.copy()
+        combined[target] = y
```

Comment:
Is the idea here to accommodate the eventuality that the input data frame contains the target variable as well as accepting duplicate target columns from y?  It seems you could just do an `X.drop(target, errors="ignore")` if you wanted to eliminate the possibility that you have duplicate target columns.  That kind of goes back to my question about being sure that the target is named "target" though.

Reply:
@chukarsten this is just to make sure that the column we add doesn't interfere with the current columns the user has. If a user has a column in the `X` data already labeled as `target`, we don't want to assume that is equivalent to the `y` data, so dropping it might be detrimental

Reply:
@chukarsten I changed the implementation for this. Thanks for the input, I think it looks much cleaner now!